packages/perseus/src/components/tooltip.tsx

```diff
@@ -40,136 +41,135 @@
-                    marginTop: -1,
-                    marginBottom: -2,
-                    zIndex: zIndex,
-                }}
-            >
-                {/* The background triangle used to create the effect of a
+    //const isRight = (this.props.horizontalDirection === "right");
+    const isTop = props.verticalDirection === "top";
+
+    const frontTopOffset = isTop ? 0 : 1;
+    const borderTopOffset = isTop ? 0 : -1;
+
+    return (
+        <div
+            style={{
+                display: "block",
+                position: props.position,
+                visibility: props.visibility,
+                left: props.left,
+                top: props["top"],
+                width: props.width + 2,
+                height: props.height + 1,
+                marginTop: -1,
+                marginBottom: -2,
+                zIndex: zIndex,
+            }}
+        >
+            {/* The background triangle used to create the effect of a
                 border around the foreground triangle*/}
-                <Triangle
-                    horizontalDirection={this.props.horizontalDirection}
-                    verticalDirection={this.props.verticalDirection}
-                    color={this.props.border}
-                    left={0}
-                    top={borderTopOffset}
-                    width={this.props.width + 2} // one extra for the diagonal
-                    height={this.props.height + 2}
-                />
-                {/* The foreground triangle covers all but the left/right edges
+            <Triangle
+                horizontalDirection={props.horizontalDirection}
+                verticalDirection={props.verticalDirection}
+                color={props.border}
+                left={0}
+                top={borderTopOffset}
+                width={props.width + 2} // one extra for the diagonal
+                height={props.height + 2}
+            />
+            {/* The foreground triangle covers all but the left/right edges
                 of the background triangle */}
-                <Triangle
-                    horizontalDirection={this.props.horizontalDirection}
-                    verticalDirection={this.props.verticalDirection}
-                    color={this.props.color}
-                    left={1}
-                    top={frontTopOffset}
-                    width={this.props.width}
-                    height={this.props.height}
-                />
-            </div>
-        );
-    },
-});
+            <Triangle
+                horizontalDirection={props.horizontalDirection}
+                verticalDirection={props.verticalDirection}
+                color={props.color}
+                left={1}
+                top={frontTopOffset}
+                width={props.width}
+                height={props.height}
+            />
+        </div>
+    );
+};
+
+TooltipArrow.defaultProps = {
+    position: "relative",
+    visibility: "visible",
+    left: 0,
+    top: 0,
+};
```

Comment:
I just saw the following code in widgets.ts:
```
    // Minor version upgrades (eg. new optional props) don't have
    // transform functions. Instead, we fill in the new props with their
    // defaults.
    const defaultProps = type in editors ? editors[type].defaultProps : {};
    newEditorProps = {
        ...defaultProps,
        ...newEditorProps,
    };
```
I guess you can ignore my earlier comment for now.  This means that functional components that are widgets must use `.defaultProps`.  I'm not sure how to enforce this.  It would be nice if we could figure out a more robust alternative to this.

Reply:
Looks like there's [`react/require-default-props`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-default-props.md), but it then forces all optional props to have a default, which isn't exactly what we want.  We may be able to fork it and adjust it to what we want.

Reply:
Nice find!  We can scope rules to different directories by adding an .eslintrc.js file in the widgets folder.